### PR TITLE
JAMES-3831 Implement urn:apache:james:params:jmap:mail:identity:sortorder extension

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala [hunk 1]
```diff
@@ -926,6 +1027,115 @@ trait IdentitySetContract {
...
+           |                        "sortOrder": 125
+           |                    }
+           |                ]
+           |            },
+           |            "c2"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+  }
+
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala:1138]: Can we add a test to check we can't modify the sortOrder without using the custom capability?

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala:1138]: Well it would be possible, however it do not hurt inter-operability (to want to modify the sort order you need to be aware it exists) => This corner case do not hurt.

But feel free to add a test if you believe it is necessary.

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Rebased to solve a merge conflict.
